server_src/runners/history.js

```diff
@@ -668,54 +639,46 @@ class HistoryCollection {
         printHistory,
       });
       await saveHistory.save().then(async (r) => {
+        const thumbnail = await HistoryCollection.thumbnailCheck(
+          payload,
+          serverSettings[0],
+          files,
+          saveHistory._id,
+          "onComplete",
+          printer
+        );
+        let snapshot = "";
```

Comment:
Had to move the if statement and create a new snapshot variable to be overwritten. Man this needs re-writing.

Reply:
Can u make issue for that?

Reply:
It's #366 for history clean up already there.